### ISSUE TITLE
Possibility of viewing pastes without javascript

### ISSUE THREAD

**Issue by Grinnz** [LINKIWI/modern-paste#28]
I personally do not disable JavaScript but many recipients of paste URLs do. The user experience without javascript enabled is nonexistent, as it is just a blank screen, so the standard paste links are unusable in such communities. It would be nice if it would display the paste without javascript initially, as people will not always be passing around the raw link to their paste, and the raw link of course is missing various metadata.

**Comment by LINKIWI** [comment]
The original motivation for this design decision was to protect the integrity of password-protected pastes, and to provide an interface for interactively unlocking a paste via a password. If the full paste metadata were shipped to the paste view template as part of rendering process, the paste contents would be always be embedded in the DOM even if no password were supplied.

To avoid the awkwardness of having to manually append a plain-text password to the URL (e.g. like `/paste/view/2?password=my-super-secret-password` for password-protected pastes, I opted for a Javascript-only solution where:

* A generic paste view template would be rendered, and we check (via Javascript) if the paste is password-protected
    * If not, the full paste metadata would be displayed in the page
    * If it is, we'll provide an interactive interface for the user to enter a password, after which the request will be repeated (via AJAX) with the password, and the metadata would be displayed in the page.

A partial solution to this would be to server-side render only non-password-protected pastes. However, this would lead to an interface disparity between regular pastes and non-password-protected pastes. On the other hand, adding the ability to include the paste password in the URL or an HTTP header is a security concern.

**Comment by Grinnz** [comment]
I agree, there is no need for password-protected pastes to be visible without javascript.

**Comment by l-n-s** [comment]
:+1: Fedora pastebin migrated to this script and it doesn't work for me now

**Comment by divinity76** [comment]
`and the raw link of course is missing various metadata.` - that could be added with http headers, although an implementation would have to be careful about the allowed characterset in http headers, for example, \r\n is _NOT_ allowed in header data, as its the http header list delimiter
